docs/source/examples/Layout and Styling.ipynb

```diff
@@ -0,0 +1,1892 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Layout and Styling of Jupyter widgets\n",
+    "\n",
+    "This notebook presents how to layout and style Jupyter interactive widgets to build rich and *reactive* widget-based applications."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## The `layout` attribute.\n",
+    "\n",
+    "Every Jupyter interactive widget has a `layout` attribute exposing a number of css properties that impact how widgets are laid out.\n",
+    "\n",
+    "### Exposed css properties\n",
+    "\n",
+    "<div class=\"alert alert-info\" style=\"margin: 20px\">\n",
+    "The following properties carry to the values of the css properties of the same name (underscores being replaced with dashes), applied to the top DOM elements of the corresponding widget.\n",
+    "</div>\n",
+    "\n",
+    "\n",
+    "** Sizes **\n",
+    "- `width`\n",
+    "- `height`\n",
+    "- `max_height`\n",
+    "- `max_width`\n",
+    "- `min_height`\n",
+    "- `min_width`\n",
+    "\n",
+    "** Display **\n",
+    "\n",
+    "- `visibility`\n",
+    "- `display`\n",
+    "- `overflow`\n",
+    "\n",
+    "** Box model **\n",
+    "- `border` \n",
+    "- `margin`\n",
+    "- `padding`\n",
+    "\n",
+    "** Positioning **\n",
+    "- `top`\n",
+    "- `left`\n",
+    "- `bottom`\n",
+    "- `right`\n",
+    "\n",
+    "** Flexbox **\n",
+    "- `order`\n",
+    "- `flex_flow`\n",
+    "- `align_items`\n",
+    "- `flex`\n",
+    "- `align_self`\n",
+    "- `align_content`\n",
+    "- `justify_content`\n",
+    "\n",
+    "\n",
+    "### Shorthand css properties\n",
+    "\n",
+    "You may have noticed that certain css properties such as `margin-[top/right/bottom/left]` seem to be missing. The same holds for `padding-[top/right/bottom/left]` etc.\n",
+    "\n",
+    "In fact, you can atomically specify `[top/right/bottom/left]` margins via the `margin` attribute alone by passing the string\n",
+    "\n",
+    "```\n",
+    "margin: 100px 150px 100px 80px;\n",
+    "```\n",
+    "\n",
+    "for a respectively `top`, `right`, `bottom` and `left` margins of  `100`, `150`, `100` and `80` pixels.\n",
+    "\n",
+    "Similarly, the `flex` attribute can hold values for `flex-grow`, `flex-shrink` and `flex-basis`. The `border` attribute is a shorthand property for `border-width`, `border-style (required)`, and `border-color`."
+   ]
```

Comment:
The segment above is great!  I'm really glad you added it because it will help a lot of people I think!